src/app/components/Rating/Rating.tsx

```diff
@@ -9,27 +9,23 @@ function Rating({
   maxRating = 5,
   className,
 }: RatingProps): JSX.Element {
-  const realRating = rating / 2;
-  rating = Math.round(realRating * 2) / 2;
   return (
     <section className={`${styles.rating} ${className}`}>
       <p className={styles.ratingValue}>{rating}</p>
       <div className={styles.starContainer}>
-        {Array.from({ length: maxRating }, (_, i) => i + 1).map(
-          (value, index) => {
-            let starStatus: 'active' | 'inactive' | 'halfActive';
-            if (value < rating + 1) {
-              if (rating % 1 === 0 || rating > value) {
-                starStatus = 'active';
-              } else {
-                starStatus = 'halfActive';
-              }
+        {Array.from({ length: maxRating }, (_, i) => i + 1).map((value) => {
```

Comment:
Very nice. But hard to understand. Would it be a good idea to write a helper function that abstract this part?:
```suggestion
        {getRangeCountFrom(maxRating).map((value) => {
```

Reply:
Cheers dude, good idea!